functions/api_flows/app.py

```diff
@@ -236,7 +234,7 @@ def put_flow_by_id(
     else:
         flow.root.created = now
     # Set these if not supplied
-    username = get_username(parse_claims(app.current_event.request_context))
+    username = app.current_event.request_context.authorizer.raw_event["username"]
```

Comment:
You have this same `if` in different places and if you need to change the logic of fields to retrieve, you will need to modify many lines. Convert it in a helper function.

Reply:
Agreed, will do

Reply:
Done